### ISSUE TITLE
[Feature] Alternative Option 

### ISSUE THREAD

**Issue by khengyun** [eli64s/readme-ai#74]
It would be better to have [g4f](https://github.com/xtekky/gpt4free) support. Alternative to GPT API Key.

**Comment by eli64s** [comment]
Hi @khengyun, thanks for sharing this! I'll take a look at implementing this into the tool.

Thank you,
*Eli*

**Comment by skfrost19** [comment]
Hi, @eli64s I would like to contribute on this feature.

**Comment by eli64s** [comment]
Hi @skfrost19. Thanks for your interest in contributing to _readme-ai_. Do you have any specific idea of how you'd like to contribute to this feature?

For additional context, the goal here is to add support for additional LLM APIs in addition to the current implementation of only the OpenAI API. This would give users more flexibility and potentially help grow the project. I need to spend more time working out the details on how to refactor the current codebase and integrate this solution. If you have any initial thoughts or ideas please go ahead and share them.

Thank you,
_Eli_

**Comment by skfrost19** [comment]
@eli64s Yeah sure, we can add 2 other options - Google Gemini API (free) and use of offline quantized models LLaMa2 or OpenChat3.5 for the text generation.
Now the offline models will take more time for the readme generation but it will get the job done. Thanks.

**Comment by eli64s** [comment]
@skfrost19 Thanks for sharing, I think integrating any free API such as Google Gemini is a good starting point.

Are you familiar with the [LiteLLM](https://github.com/BerriAI/litellm) project? It allows you to interface with many providers using the same methods that OpenAI uses. I need to spend some more time learning about LiteLLM but it may be an interesting option to explore as well.

**Comment by skfrost19** [comment]
@eli64s i am not familiar with the project but will look into it and get back to you.
Thanks

**Comment by wywerne** [comment]
another open source project : https://jan.ai/
